### DISCUSSION TITLE
Implicit conversion from float

### DISCUSSION THREAD

**Discussion by seabasss** [spatie/image#193] [General]
Any way to get around this?

```Implicit conversion from float 1223.111111111111 to int loses precision in /vendor/spatie/image/src/Manipulations.php on line 156```

Thank you!

**Comment by bdahlinger** [comment]
Started seeing this too, haven't started to look into it yet but possibly related to something fixed in another of their packages : 
 https://github.com/spatie/color/releases/tag/1.3.2

**Reply by bdahlinger** [reply]
After spending some time on this, I had upgraded site to php 8.1 which now sends warnings if you try and pass in floats when expecting ints on the Manipulation class line 146 (new to php8.1). This issue is in spatie image, but it's probably same issue I have with spatie media library which of course uses this package. To fix, I've changed my media models to handle this e.g., `(int) round($width) `  in the `->manualCrop`. Then ran a seeder to convert all json custom_properties for the crop to integers. This stopped the warnings which were 4000 + for me, couldn't use spatie Ray because it was taking over it so to speak on any request. Hope this helps!

**Reply by seabasss** [reply]
Thanks! That's what I ended up doing! It's asking for int so it was my fault :)
